tests/js/spec/utils/withRepositories.spec.jsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+
+import {mount} from 'sentry-test/enzyme';
+
+import RepositoryStore from 'app/stores/repositoryStore';
+import withRepositories from 'app/utils/withRepositories';
+
+describe('withRepositories HoC', function() {
+  const orgSlug = 'myOrg';
+  const repoUrl = `/organizations/${orgSlug}/repos/`;
+
+  const api = new MockApiClient();
+  const mockData = [{id: '1'}];
+
+  beforeEach(() => {
+    MockApiClient.clearMockResponses();
+    MockApiClient.addMockResponse({
+      url: repoUrl,
+      body: mockData,
+    });
+
+    jest.restoreAllMocks();
+    RepositoryStore.init();
+  });
+
+  it('adds repositories prop', async () => {
+    const Component = () => null;
+    const Container = withRepositories(Component);
+    const wrapper = mount(<Container api={api} orgSlug={orgSlug} />);
+
+    await tick(); // Run Store.loadRepositories
+    await tick(); // Run Store.loadRepositoriesSuccess
+    wrapper.update(); // Re-render component with Store data
+
+    const mountedComponent = wrapper.find(Component);
+    expect(mountedComponent.prop('repositories')).toEqual(mockData);
+    expect(mountedComponent.prop('repositoriesLoading')).toEqual(false);
+    expect(mountedComponent.prop('repositoriesError')).toEqual(undefined);
+  });
+
+  it('prevents repeated calls', async () => {
+    const Component = () => null;
+    const Container = withRepositories(Component);
+
+    // XXX(leedongwei): We cannot spy on `fetchRepositories` as Jest can't
+    // replace the method in the prototype due to createReactClass.
+    // As such, I'm using `componentDidMount` as a proxy.
```

Comment:
:/ alternatively, we have this pattern we can use with es6 classes: https://github.com/getsentry/sentry/blob/464fa864c4f3434aca55e974bb26cf8977e99f3e/src/sentry/static/sentry/app/views/organizationGroupDetails/groupDetails.tsx#L170

But this is fine too